test/system/src/test/java/io/pravega/test/system/SingleSubscriberUpdateRetentionStreamCutTest.java

```diff
@@ -0,0 +1,209 @@
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class SingleSubscriberUpdateRetentionStreamCutTest extends AbstractReadWriteTest {
+
+    private static final String SCOPE = "testCBRScope" + RandomFactory.create().nextInt(Integer.MAX_VALUE);
+    private static final String STREAM = "testCBRStream" + RandomFactory.create().nextInt(Integer.MAX_VALUE);
+    private static final String READER_GROUP = "testCBRReaderGroup" + RandomFactory.create().nextInt(Integer.MAX_VALUE);
+    private static final String SIZE_30_EVENT = "data of size 30";
+
+    private static final int READ_TIMEOUT = 1000;
+    private static final int MAX_SIZE_IN_STREAM = 300;
+    private static final int MIN_SIZE_IN_STREAM = 30;
+
+    private final ReaderConfig readerConfig = ReaderConfig.builder().build();
+    private final ScheduledExecutorService executor = ExecutorServiceHelpers.newScheduledThreadPool(4, "executor");
+    private final ScheduledExecutorService streamCutExecutor = ExecutorServiceHelpers.newScheduledThreadPool(1, "streamCutExecutor");
+    private URI controllerURI = null;
+    private StreamManager streamManager = null;
+    private Controller controller = null;
+
+    /**
+     * This is used to setup the various services required by the system test framework.
+     *
+     * @throws MarathonException    when error in setup
+     */
+    @Environment
+    public static void initialize() throws MarathonException {
+        URI zkUri = startZookeeperInstance();
+        startBookkeeperInstances(zkUri);
+        URI controllerUri = ensureControllerRunning(zkUri);
+        ensureSegmentStoreRunning(zkUri, controllerUri);
+    }
+
+    @Before
+    public void setup() {
+        Service conService = Utils.createPravegaControllerService(null);
+        List<URI> ctlURIs = conService.getServiceDetails();
+        controllerURI = ctlURIs.get(0);
+
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+
+        controller = new ControllerImpl(ControllerImplConfig.builder()
+                .clientConfig(clientConfig)
+                .maxBackoffMillis(5000).build(), executor);
+        streamManager = StreamManager.create(clientConfig);
+
+        assertTrue("Creating scope", streamManager.createScope(SCOPE));
+        assertTrue("Creating stream", streamManager.createStream(SCOPE, STREAM,
+                StreamConfiguration.builder()
+                        .scalingPolicy(ScalingPolicy.fixed(1))
+                        .retentionPolicy(RetentionPolicy.bySizeBytes(MIN_SIZE_IN_STREAM, MAX_SIZE_IN_STREAM)).build()));
+    }
+
+    @After
+    public void tearDown() {
+        streamManager.close();
+        ExecutorServiceHelpers.shutdown(executor);
```

Comment:
You have actually missed to shut down the second executor.

Reply:
I added the shutdown